### PR TITLE
Watch blog post

### PR DESCRIPTION
**Pull request by lukehoban**

### CODE REVIEW

File: content/blog/pulumi-watch-mode-fast-inner-loop-development-for-cloud-infrastructure/index.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: "Pulumi Watch Mode - Fast Inner Loop Development for Cloud Infrastructure"
+authors: ["luke-hoban"]
+tags: ["serverless","kubernetes","Logging","New Features","applications"]
+date: "2019-12-02"
+meta_desc: "Pulumi Watch provides a mode for developing cloud infrastructure that speeds up the rate of iteration and allows cloud developers to focus on their code and infrastructure, instead of on the mechanics of their deployments, across modern cloud platforms including serverless, Kubernetes and more."
+meta_image: "unnamed.png"
```
Comment (lukehoban) [inline content/blog/pulumi-watch-mode-fast-inner-loop-development-for-cloud-infrastructure/index.md]: A real meta image is needed before posting - working on that now.

File: content/blog/pulumi-watch-mode-fast-inner-loop-development-for-cloud-infrastructure/index.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+This new `watch` mode isn’t a replacement for a standard `pulumi preview`/`pulumi update`.  For production infrastructure deployments, it is critical to understand precisely what changes will happen before applying them, to integrate with CI/CD systems, to be able to do targeted updates, and to apply policy to guard the production environment against misconfiguration.  But that same level of care isn’t necessarily the right way to approach building new features in a development stack.  And it’s in these development environments that `pulumi watch` opens up a new experience for infrastructure development.  Although they are shorter periods relative to the full life of an infrastructure deployment, they are the majority of the time that creativity and creation is applied to cloud infrastructure, and thus represent a key opportunity to empower developers.
```
Comment (justinvp) [inline content/blog/pulumi-watch-mode-fast-inner-loop-development-for-cloud-infrastructure/index.md]: Add a link to the CrossGuard blog post around "apply policy"?

```suggestion
This new `watch` mode isn’t a replacement for a standard `pulumi preview`/`pulumi update`.  For production infrastructure deployments, it is critical to understand precisely what changes will happen before applying them, to integrate with CI/CD systems, to be able to do targeted updates, and to [apply policy]({{< relref "/blog/announcing-crossguard-preview" >}}) to guard the production environment against misconfiguration.  But that same level of care isn’t necessarily the right way to approach building new features in a development stack.  And it’s in these development environments that `pulumi watch` opens up a new experience for infrastructure development.  Although they are shorter periods relative to the full life of an infrastructure deployment, they are the majority of the time that creativity and creation is applied to cloud infrastructure, and thus represent a key opportunity to empower developers.
```

File: content/blog/pulumi-watch-mode-fast-inner-loop-development-for-cloud-infrastructure/index.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+Not all cloud resources can be deployed or updated in seconds - but many of the most popular modern cloud primitives are built on platforms that enable and encourage quick iteration - and in particular both serverless and Kubernetes.  Watch mode can be used during the development of any infrastructure, but can really shine in these areas where the cloud platforms natively support fast iteration.
+
+## Kubernetes Infrastructure + Applications
+
+For Kubernetes, `pulumi watch` allows us to deploy changes to our cluster automatically as we author our Kubernetes resources.  Change the number of `replicas` for our `Deployment` and hit save and the new pods scale-out.  Add a `Service` to expose the deployment and public endpoint is available immediately.  This pairs beautifully with the recently released Pulumi Kubernetes Extensions (`kubernetesx`) library, which allows you to build Kubernetes applications with just a few lines of code - avoiding a lot of the boilerplate that slows down typical Kubernetes development when working with YAML.  We can see this combination in action in the video below:
```
Comment (justinvp) [inline content/blog/pulumi-watch-mode-fast-inner-loop-development-for-cloud-infrastructure/index.md]: Link to kubernetesx library on GitHub?

File: content/blog/pulumi-watch-mode-fast-inner-loop-development-for-cloud-infrastructure/index.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+{{< youtube "FuYwsXLqnTk?rel=0" >}}
+
+# Conclusion
+
+When using `pulumi watch`, working with cloud infrastructure *feels* so different - it feels malleable, it feels accessible, and it feels like it’s right at your fingertips.  By building on the same robust incremental deployment infrastructure as the rest of Pulumi, you get the same access to any cloud, and to a variety of languages and all the other benefits of Pulumi’s modern infrastructure as code tooling.  Pulumi watch triggers a deploy on any save - including both of files in the Pulumi program itself, as well as other assets that the Pulumi program may reference like application source code or static assets.  Pulumi watch also displays [runtime logs](https://www.pulumi.com/blog/unified-logs-with-pulumi-logs/) for supported platforms (Lambda, ECS, and Cloudwatch currently, more coming soon), and can be paired with another new feature, [Pulumi Query](https://www.pulumi.com/blog/query-kubernetes/), to get customized status about the runtime environment during rapid iteration.
```
Comment (justinvp) [inline content/blog/pulumi-watch-mode-fast-inner-loop-development-for-cloud-infrastructure/index.md]: Use `relref` when linking inside the site.

```suggestion
When using `pulumi watch`, working with cloud infrastructure *feels* so different - it feels malleable, it feels accessible, and it feels like it’s right at your fingertips.  By building on the same robust incremental deployment infrastructure as the rest of Pulumi, you get the same access to any cloud, and to a variety of languages and all the other benefits of Pulumi’s modern infrastructure as code tooling.  Pulumi watch triggers a deploy on any save - including both of files in the Pulumi program itself, as well as other assets that the Pulumi program may reference like application source code or static assets.  Pulumi watch also displays [runtime logs]({{< relref "/blog/unified-logs-with-pulumi-logs" >}}) for supported platforms (Lambda, ECS, and Cloudwatch currently, more coming soon), and can be paired with another new feature, [Pulumi Query]({{< relref "/blog/query-kubernetes" >}}), to get customized status about the runtime environment during rapid iteration. 
```

File: content/blog/pulumi-watch-mode-fast-inner-loop-development-for-cloud-infrastructure/index.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+The `pulumi watch` command is available in preview now when the `PULUMI_EXPERIMENTAL` environment variable is set to `true`.  We are just getting started - and see many more opportunities to tighten the iteration cycle further, to support more logging capabilities, to integrate more deeply with `query`, and to be applied to more modern infrastructure scenarios.  
+
+Give it a try today and drop us a note about where you’d like to see this go for [Pulumi 2.0](https://www.pulumi.com/blog/pulumi-2-0-roadmap/)!
```
Comment (justinvp) [inline content/blog/pulumi-watch-mode-fast-inner-loop-development-for-cloud-infrastructure/index.md]:
```suggestion
Give it a try today and drop us a note about where you’d like to see this go for [Pulumi 2.0]({{< relref "/blog/pulumi-2-0-roadmap" >}})!
```

**Pull request merged**